src/pool.rs

```diff
@@ -149,11 +230,37 @@ impl ConnectionPool {
         }
     }
 
-    // Connects to a MongoDB server as defined by the initial configuration.
-    fn connect(&self) -> Result<BufStream<TcpStream>> {
+
+    // Connects to a MongoDB server as defined by the initial
+    #[allow(unreachable_code)] // Suppresses warning for `panic` when ssl is enabled.
+    fn connect(&self) -> Result<BufStream<Stream>> {
         let host_name = &self.host.host_name;
         let port = self.host.port;
-        let stream = BufStream::new(try!(TcpStream::connect((&host_name[..], port))));
-        Ok(stream)
+        let inner_stream = try!(TcpStream::connect((&host_name[..], port)));
+
+        if let Some(SslConfig { ca_file: ref _ca_file,
+                                certificate_file: ref _certificate_file,
+                                key_file: ref _key_file }) = self.ssl {
+            #[cfg(feature = "ssl")]
+            {
+                let mut ssl_context = SslContext::builder(SslMethod::tls())?;
+                ssl_context.set_cipher_list("DEFAULT")?;
```

Comment:
From looking at the options that the mongo shell has, the SSL verification options seem to be `--sslAllowInvalidHostnames` and `--sslAllowInvalidCertificates` (which both default to off), whereas the OpenSSL crate has `SSL_VERIFY_NONE`, SSL_VERIFY_PEER`, and `SSL_VERIFY_FAIL_IF_NO_PEER_CERT`. If I correctly understand the OpenSSL crate documetation on these types correctly, then `SSL_VERIFY_PEER` and `SSL_VERIFY_FAIL_IF_NO_PEER_CERT` should be on by default, and `--sslAllowInvalidCertificates` is the equivalent of turning off `SSL_VERIFY_PEER` (but not `SSL_VERIFY_FAIL_IF_NO_PEER_CERT`. I'm having trouble finding the equivalent option to `--sslAllowInvalidHostnames` in the OpenSSL crate, but I'd be all right with leaving out the ability to use this option unless/until someone asks to be able to use it.

The cipher list is a bit trickier; I'm having trouble finding documentation specifying exactly which ciphers are able to be used in MongoDB, but from looking at the options available to `mongo` and `mongod`, I think this is something that's only configurable on the server-side, so this might just be something that drivers don't have a specific API for. I'm going to be back at work tommorrow, so I'll check with some of the drivers people tomorrow to see what the other drivers do with regard to this.